widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/Toolbar.java

```diff
@@ -74,52 +74,56 @@ public Toolbar() {
     super.add( bar );
     super.setCellWidth( bar, "100%" );
     setWidth( "100%" ); //$NON-NLS-1$
-    Roles.getToolbarRole().set( bar.getElement() );
-    this.addKeyDownHandler( event -> {
-      Element activeElement = Element.as( event.getNativeEvent().getEventTarget() );
-      Element nextElement = null;
-      switch ( event.getNativeKeyCode() ) {
-        case KeyCodes.KEY_RIGHT:
-          nextElement = ElementUtils.findNextKeyboardFocusableElement( activeElement, bar.getElement() );
-          break;
-        case KeyCodes.KEY_LEFT:
-          nextElement = ElementUtils.findPreviousKeyboardFocusableElement( activeElement, bar.getElement() );
-          break;
-        case KeyCodes.KEY_HOME:
-          nextElement = ElementUtils.findFirstKeyboardFocusableDescendant( bar.getElement() );
-          break;
-        case KeyCodes.KEY_END:
-          nextElement = ElementUtils.findLastKeyboardFocusableDescendant( bar.getElement() );
-          break;
-        case KeyCodes.KEY_TAB:
-          Element node;
-          if ( event.isShiftKeyDown() ) {
-            node = ElementUtils.findFirstKeyboardFocusableDescendant( bar.getElement() );
-            ElementUtils.tabPrevious( node );
-          } else {
-            node = ElementUtils.findLastKeyboardFocusableDescendant( bar.getElement() );
-            ElementUtils.tabNext( node );
-          }
-          break;
-        default:
-          return;
+  }
+
+  @Override
+  protected void onAttach() {
+    super.onAttach();
+
+    jsToolbarController = createJsToolbarController( this, getElement() );
+  }
+
+  @Override
+  protected void onDetach() {
+    if ( jsToolbarController != null ) {
+      destroyJsToolbarController( jsToolbarController );
+    }
+
+    super.onDetach();
+  }
+
+  private static native JavaScriptObject createJsToolbarController( Toolbar toolbar, Element container )/*-{
+    return $wnd.pho.util._a11y.makeAccessibleToolbar( container, {
+      itemSelector: ".toolbar-button-focus-panel",
+      itemFilter: function(itemElem) {
+        return toolbar.@org.pentaho.gwt.widgets.client.toolbar.Toolbar::itemFilter(Lcom/google/gwt/dom/client/Element;)( itemElem );
       }
+    });
+  }-*/;
 
-      event.preventDefault();
 
-      if ( nextElement != null ) {
-        nextElement.focus();
+  private boolean itemFilter( Element element ) {
+    boolean result = false;
```

Comment:
Imo, no need for variable.
It's preferable to return the result from inside the if. And return false at the end.